openvpn/middlewares/server/filter/middleware.go

```diff
@@ -42,60 +42,39 @@ END
 
 var filterLAN = template.Must(template.New("filter_lan").Parse(filterLANTemplate))
 
+// Exposes API to control client's packet filtering.
+//
+// The OpenVPN server should have been started with the
+// --management-client-pf directive so that it will require that
+// VPN tunnel packets sent or received by client instances must
+// conform to that client's packet filter configuration.
 type middleware struct {
+	*auth.Middleware
```

Comment:
Why `filter` depends on `auth`?

Reply:
Simple inheritances. `auth.Middleware` general purpose - subscribes to client status events

Reply:
There is no inheritance for golang, that is why it looks weird when you are trying to invent it yourself.